packages/api/src/utils/billing.js

```diff
@@ -1,24 +1,40 @@
 /* eslint-disable no-void */
 
+import { InvalidTosAgreementError, TosAgreementRequiredError } from '../../src/errors.js'
+
 /**
  * @typedef {import('./billing-types').StoragePriceName} StoragePriceName
  */
 
+/**
+ * @param {import('./billing-types').Agreement} agreement
+ * @returns {boolean}
+ */
+function isAgreement (agreement) {
+  return agreement === 'web3.storage-tos-v1'
+}
+
 /**
  * Save a user's payment settings
  * @param {object} ctx
  * @param {import('./billing-types').BillingService} ctx.billing
  * @param {import('./billing-types').CustomersService} ctx.customers
  * @param {import('./billing-types').SubscriptionsService} ctx.subscriptions
+ * @param {import('./billing-types').TermsOfServiceService} ctx.termsOfService
  * @param {import('./billing-types').BillingUser} ctx.user
  * @param {object} paymentSettings
  * @param {Pick<import('./billing-types').PaymentMethod, 'id'>} paymentSettings.paymentMethod
  * @param {import('./billing-types').W3PlatformSubscription} paymentSettings.subscription
+ * @param {import('./billing-types').Agreement} paymentSettings.agreement
  * @param {import('./billing-types').UserCreationOptions} [userCreationOptions]
  */
 export async function savePaymentSettings (ctx, paymentSettings, userCreationOptions) {
-  const { billing, customers, user } = ctx
+  if (!paymentSettings.agreement) throw new TosAgreementRequiredError()
+  if (!isAgreement(paymentSettings.agreement)) throw new InvalidTosAgreementError()
+
+  const { billing, customers, user, termsOfService } = ctx
   const customer = await customers.getOrCreateForUser(user, userCreationOptions)
+  await termsOfService.createUserTosAgreement(user.id, paymentSettings.agreement)
```

Comment:
in practice the calls to customers and billing are going to hit the stripe apis so be slow compared to the ones that just use the sql db. Don't have to do it now, but eventually we might want to do some stuff in parallel here? I think technically there are three parallelizable steps. wdyt in theory? Don't have to do now but good to have in mind if this starts feeling slow.

```
function 
await Promise.all([
  termsOfService.createUserTosAgreement(user.id, paymentSettings.agreement),
  billing.savePaymentMethod(customer.id, paymentSettings.paymentMethod.id),
  (async () => {
    const customer = await customers.getOrCreateForUser(user, userCreationOptions);
    await ctx.subscriptions.saveSubscription(customer.id, paymentSettings.subscription)
  })(),
]);
```